store/src/java/com/zimbra/cs/filter/RuleManager.java

```diff
@@ -164,6 +175,81 @@ private static void setRules(Entry entry, String script, String sieveScriptAttrN
         }
     }
 
+    /**
+     * Reject the sieve script by throwing a ServiceException if its size exceeds
+     * the limit in the {@code zimbraMailSieveScriptMaxSize} attribute. If the new sieve
+     * script exceeds the limit, but is smaller than before, then the new script is not
+     * rejected.
+     *
+     * If the value of the attribute is not set, or if the attribute getter is not
+     * defined for the entry type, then the limit defined in the global config is used.
+     *
+     * @param entry the account/domain/cos/server for which the rules are to be saved
+     * @param newSieveScript the new sieve script, or an empty string
+     * @param sieveScriptAttrName the name of the sieve script attribute
+     * @throws ServiceException if the sieve script exceeds the value of the max size attribute
+     */
+    private static void rejectSieveScriptExceedingMaxSize(Entry entry, String newSieveScript, String sieveScriptAttrName)
+            throws ServiceException {
+        long sieveScriptMaxSize;
+        String zimbraMailSieveScriptMaxSize = entry.getAttr(Provisioning.A_zimbraMailSieveScriptMaxSize);
```

Comment:
`<attr id="9001" name="zimbraMailSieveScriptMaxSize" type="long" min="0" cardinality="single" optionalIn="globalConfig,account,cos,domain" flags="domainInfo,accountInfo,accountCosDomainInherited" since="11.0.0">`
The attribute can have a value on globalConfig, domain, cos and account levels. If no value is set on domain, cos or account, the value on globalConfig should be used. I noticed, however, `domainInherited` flag has not been set.
Please add the flag in zimbra-attrs.xml and verify that `domainInherited` and `accountCosDomainInherited` work as expected.
```
(zimbra-attrs.xml)
    domainInherited........if not set on domain, inherit from global config
    accountCosDomainInherited...if not set on account, inherit from COS,
                                if not set on COS, inherit from domain
```